rotkehlchen/chain/ethereum/modules/yearn/decoder.py

```diff
@@ -58,14 +62,18 @@ def __init__(
             base_tools=base_tools,
             msg_aggregator=msg_aggregator,
         )
+        self.vaults = EMPTY_VAULTS
 
+    def reload_data(self) -> None:
+        """Reload vaults from DB."""
+        self.vaults = EMPTY_VAULTS
```

Comment:
have you checked if this works? I'm under the impression that the one in the init

```
        self.vaults = EMPTY_VAULTS
```

will create a reference to `EMPTY_VAULTS` and when you populate it you will be actually modifying `EMPTY_VAULTS` so when you do this you will still have everythin in memory.

It's not a problem because it is a set and you are doing .add to the set. In fact I believe here you don't need to reset so in the init method you can do

```
self.vaults = {
    YEARN_VAULTS_V1_PROTOCOL: set(),
    YEARN_VAULTS_V2_PROTOCOL: set(),
}
```